packages/pieces/community/dust/src/lib/actions/upload-file.ts

```diff
@@ -0,0 +1,43 @@
+import { createAction, Property } from '@activepieces/pieces-framework';
+
+import { createClient } from '../common';
+import { dustAuth, DustAuthType } from '../..';
+import mimeTypes from 'mime-types';
+import { FileUploadUrlRequestType } from '@dust-tt/client';
+
+export const uploadFile = createAction({
```

Comment:
What's the difference with "add fragment"?

Reply:
Uploading a file is not tied to a conversation. It's tied to the workspace and you can refer the file in further conversation. That's why I decided to not rely on the existing "add fragment" piece.

Do you think I should do differently?

Reply:
Ah got it - I think it's rather "new"
If it's the same API endpoint, one alternative would be to have a single action but make the "conversation ID" prop optional
If it's a separate concept / endpoint, let's keep 2 actions (but that means you have fragments, files, and documents 🤯 )

Reply:
it's new yes. I think fragment will disappear overtime. Let's keep separate for now as the endpoints are different